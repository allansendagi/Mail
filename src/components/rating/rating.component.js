import React from 'react';
import './rating.styles.scss';
import {Icon, Progress} from 'semantic-ui-react';

export function Rating(props) {

	let progress = null;
	if (props.likeCount && props.dislikeCount) {
		const percent =100 * (props.likeCount / (props.likeCount + props.dislikeCount));
		Progress = <progress className='progress' percent={percent} size='tiny' />;
	}

	return(
		<div className='rating'>
		  <div className='thumbs-up'>
		  	<Icon name='thumbs outline up' />
		  	 <span>{props.likeCount}</span>
		  </div>
		  <div className='thumbs-down'>
		   <Icon name='thumbs outline down' />
		   <span>{props.dislikeCount}</span>	
		  </div>
		  {Progress}	
		</div>
	)
}