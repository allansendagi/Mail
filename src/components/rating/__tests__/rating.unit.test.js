import React from 'react';
import {shallow, configure} from 'enzyme';
import {Rating} from '../rating.component';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
describe('Rating', () => {
  test('renders', () => {
    const wrapper = shallow(<Rating/>);
    expect(wrapper).toMatchSnapshot();
  });
});