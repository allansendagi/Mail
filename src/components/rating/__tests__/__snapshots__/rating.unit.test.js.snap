// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders 1`] = `ShallowWrapper {}`;
